File: frontend/types/checkin.ts
// 签到信息
export interface CheckInInfo {
  id: number
  checkin_id: string
  user_id: number
  nickname: string
  description: string
  latitude: number
  longitude: number
  location_name: string
  distance: number
  created_at: string
  tags: string[]
  likes_count: number
  comments_count: number
  liked_by_me: boolean
}

// 创建签到请求
export interface CreateCheckinRequest {
  description: string
  latitude: number
  longitude: number
  location_name: string
  tags?: string[]
}

// 创建签到响应
export interface CreateCheckinResponse {
  checkin_id: string
}

// 删除签到请求
export interface DeleteCheckinRequest {
  checkin_id: string
}

// 删除签到响应
export interface DeleteCheckinResponse {
  success: boolean
}

// 查询签到详情请求
export interface SearchCheckInByIdRequest {
  checkin_id: string
}

// 按位置搜索签到请求
export interface SearchCheckinsByLocationRequest {
  latitude: number
  longitude: number
  radius: number
  cursor?: number
  limit?: number
}

// 查询用户签到历史请求
export interface UserCheckinsHistoryRequest {
  user_id?: number
  cursor?: number
  limit?: number
}

// 按标签搜索签到请求
export interface SearchCheckinsByTagsRequest {
  tags: string[]
  latitude: number
  longitude: number
  cursor?: number
  limit?: number
} 

File: backend/src/models/api/checkin.rs
use crate::utils::response::{PaginatedResponse, Pagination};
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

// 签到API请求/响应模型
#[derive(Debug, Serialize, Deserialize)]
pub struct CreateCheckinRequest {
    pub description: String,
    pub latitude: f64,
    pub longitude: f64,
    pub location_name: String,
    pub tags: Option<Vec<String>>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateCheckinResponse {
    pub checkin_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct DeleteCheckinRequest {
    pub checkin_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct DeleteCheckinResponse {
    pub success: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchCheckInByIdRequest {
    pub checkin_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchCheckinsByLocationRequest {
    pub latitude: f64,
    pub longitude: f64,
    pub radius: f64,
    #[serde(flatten)]
    pub pagination: Pagination,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UserCheckinsHistoryRequest {
    pub user_id: Option<i64>,
    #[serde(flatten)]
    pub pagination: Pagination,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchCheckinsByTagsRequest {
    pub tags: Vec<String>,
    pub latitude: f64,
    pub longitude: f64,
    #[serde(flatten)]
    pub pagination: Pagination,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LikeCheckinRequest {
    pub checkin_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LikeCheckinResponse {
    pub success: bool,
    pub likes_count: i64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UnlikeCheckinRequest {
    pub checkin_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UnlikeCheckinResponse {
    pub success: bool,
    pub likes_count: i64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CheckInInfo {
    pub id: i64,
    pub checkin_id: String,
    pub user_id: i64,
    pub nickname: String,
    pub description: Option<String>,
    pub latitude: f64,
    pub longitude: f64,
    pub location_name: String,
    pub distance: f64,
    pub created_at: DateTime<Utc>,
    pub tags: Vec<String>,
    pub likes_count: i64,
    pub comments_count: i64,
    pub liked_by_me: bool,
}

pub type UserCheckinsHistoryResponse = PaginatedResponse<CheckInInfo>;
pub type SearchCheckinByLocationResponse = PaginatedResponse<CheckInInfo>;
pub type SearchCheckinsByTagsResponse = PaginatedResponse<CheckInInfo>;
